src/passes/OptimizeInstructions.cpp

```diff
@@ -1646,16 +1670,16 @@ struct OptimizeInstructions
 
   // TODO: templatize on type?
   Expression* optimizeRelational(Binary* binary) {
-    // TODO: inequalities can also work, if the constants do not overflow
     auto type = binary->right->type;
-    // integer math, even on 2s complement, allows stuff like
-    // x + 5 == 7
-    //   =>
-    //     x == 2
     if (binary->left->type.isInteger()) {
+      // TODO: inequalities can also work, if the constants do not overflow
+      // integer math, even on 2s complement, allows stuff like
```

Comment:
This sentence ends abruptly and it's not clear that the comment below intends to continue it, so it would be good to move the comment below to here.